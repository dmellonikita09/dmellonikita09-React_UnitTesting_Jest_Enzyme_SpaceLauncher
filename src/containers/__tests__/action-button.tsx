import React from 'react';
import ActionButton from '../action-button';
import { mount, shallow } from 'enzyme';

describe('Action Button', () => {
  it('renders without error', () => {
    const wrapper = mount(<ActionButton />);
    wrapper.find('button').simulate('click');

  });

  it("renders correctly", () => {
    const wrapper = shallow(
      <ActionButton />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('shows add trip ', () => {
    const wrap = shallow(
      <ActionButton id="1" />)
    expect(wrap.text()).toEqual('<ToggleTripButton />')
  });

  it('shows cancels trip if trip is already booked', () => {
    const wrap = shallow(
      <ActionButton isBooked={true} />
    )
    expect(wrap.text()).toEqual('<CancelTripButton />')
  });
});
