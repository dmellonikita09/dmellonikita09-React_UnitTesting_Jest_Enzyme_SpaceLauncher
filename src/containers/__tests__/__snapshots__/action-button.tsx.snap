// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Action Button renders correctly 1`] = `<ToggleTripButton />`;
